doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,449 @@
+# Solve an equation algebraically
+
+Use SymPy to solve an equation algebraically (symbolically). For example, solving $x^2 = y$ for $x$ yields $x \in \{-\sqrt{y},\sqrt{y}\}$.
+
+Alternatives to consider:
+- SymPy can also [solve many other types of problems including sets of equations](index.md).
+- Some equations cannot be solved algebraically (either at all or by SymPy), 
+so you may have to {func}`solve your equation numerically <sympy.solvers.solvers.nsolve>` instead.
+
+There are two high-level functions to solve equations, {func}`~.solve` and {func}`~.solveset`.
+Here is a simple example of each:
+
+{func}`~.solve`
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solution = solve(x ** 2 - y, x, dict=True)
+>>> print(solution)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+```
+
+{func}`~.solveset`
+
+```py
+>>> from sympy import solveset
+>>> from sympy.abc import x, y
+>>> solution = solveset(x**2 - y, x)
+>>> print(solution)
```

Comment:
I thought each code block was tested in a separate context, so I thought each had to import all packages it used. But now, as I understand it, the whole page is tested in a single context, so that's not necessary. As a new user of a package, though, I appreciate being able to simply click the copy button on a code block, paste it into my editor, and know that it will run--it can be difficult to figure out which sub-module to import from. So I think it's worth repeating the couple of lines to make it easier on new users. But if the community consensus is to remove repetitive imports, I can do that.

Similarly, I thought I had to use `print` statements because outputs aren't produced in my development environment from e.g. 

```py
>>> solveset(x**2 - y, x)
```

But I understand that doctest will print such results, so I've removed many `print` statements.

Reply:
> outputs aren't produced in my development environment from e.g.
> 
> ```python
> >>> solveset(x**2 - y, x)
> ```

What "development environment" are you using? If you run that in any ordinary Python REPL you will see some output.

Reply:
I'm using Visual Studio Code. If I run a script there in Run and Debug mode, or execute the script from the terminal, it does not produce output on such steps, which I'll term implicit output. If I run in the IDLE Shell, it does produce implicit output. So I appreciate your point that doctests run more like a REPL than a script.

I find it easier to put the code in a file and then run it to ensure that the code in the script has everything it needs (imports) to run. Contrarily, if I develop and run code interactively in a REPL, I might not copy into the docstring some line that I forgot was needed to make the code work.

I discovered that VS Code has interactive commands where you can right-click code in a file and select e.g. Run Current File in Interactive Window. That creates an interactive window which behaves like to a Jupyter Notebook. It works pretty well for this purpose; a drawback is that it only produces implicit output from the last line in the script. It is possible to produce implicit output from a previous line by right-clicking on it and selecting Run Selection/Line in Interactive Window. So I'll plan to use those features from now on.

Reply:
Repeating imports is fine when it's not clear that the person reading the example will have read the previous examples where the original imports are. They're not needed but they can help. I do agree we should just use `solveset(x**2 - y, x)` and let it show the output instead of `print(solution)`.

Reply:
Thanks. I did remove the unnecessary `print` statements, now that I understand doctests are run in a REPL.